src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs

```diff
@@ -90,46 +95,46 @@ public override int Next()
 
             public override int Next(int maxValue)
             {
-                if (maxValue > 1)
+                Debug.Assert(maxValue >= 0);
```

Comment:
It looks like this implementation could just be `public override int Next(int maxValue) => Next(0, maxValue);`, and effectively the only additional cost would be one subtraction and one addition.  Can we avoid the duplication by doing so?  Is the difference measurable?

Reply:
Yes, this could be done. As I recall it added around 0.3 ns to the benchmarks for me. Seemed important at the time. Not so much anymore. Should it be changed?

Reply:
Actually, we should be able to write just two static utility methods, one that takes a uint max and one that takes a ulong max, and then use those two routines from the four methods being modified here across both version of xoshiro. There shouldn't be any additional cost if those two helpers are aggressive inlining.

Reply:
The methods should also be able to access the generator methods - otherwise the cost increased. Tried passing the generator methods as arguments to static methods, but this appeared to be even more expensive.

I have now added a common abstract base class with the implementation logic. On my system this gives approximately the same results as before. Will this be an acceptable solution?

I guess one could discuss the layers of the classes. I went with minimal intervention to make it work, and hence putting LemireImpl as base to XoshiroImpl.

Reply:
Thanks.  I was thinking more like this: https://github.com/dotnet/runtime/compare/main...stephentoub:runtime:randomrange